bootable-jar/boot/src/main/java/org/wildfly/core/jar/boot/Main.java

```diff
@@ -190,13 +208,20 @@ private static void unzip(InputStream wf, Path dir) throws Exception {
                         Files.createDirectories(parent);
                     }
                     Files.copy(zis, newFile, StandardCopyOption.REPLACE_EXISTING);
+                    if (!isWindows && newFile.getFileName().toString().endsWith(".sh")) {
+                        Files.setPosixFilePermissions(newFile, EXECUTE_PERMISSIONS);
```

Comment:
Just a general FYI we may want to record for later, the method setPosixFilePermissions involves a SecurityManager permissions check - HOWEVER this Main class is the entry point to the application meaning it will be the only protection domain on the call stack, all a doPrivileged call does is drop the parent protection domains from the call stack so as this is class is the first this is not nescesary.